### PR TITLE
Feature/red 109 exams fields tom

### PR DESCRIPTION
**Pull request by red-tt-bear**
I did a few tests in the playground using this code:
```typescript
    try {
        await courseRepository.createStudentTopicOverride({
            courseTopicContentId: 1,
            deadDate: new Date(),
            endDate: moment().add(1, 'days').toDate(),
            startDate: moment().subtract(1, 'days').toDate(),
            userId: 1,
        });    
    } catch (e) {
        throw new WrappedError('createStudentTopicOverride Success case', e);
    }

    try {
        await courseRepository.createStudentTopicOverride({
            userId: 1,
            courseTopicContentId: 1,
        });    
    } catch (e) {
        throw new WrappedError('createStudentTopicOverride Success case no fields', e);
    }

    try {
        await courseRepository.createStudentTopicOverride({
            userId: -1,
            courseTopicContentId: 1,
        });
        throw new WrappedError('createStudentTopicOverride FK 1');
    } catch (e) {
    }

    try {
        await courseRepository.createStudentTopicOverride({
            userId: 1,
            courseTopicContentId: -1,
        });
        throw new WrappedError('createStudentTopicOverride FK 2');
    } catch (e) {
    }

    try {
        await courseRepository.createStudentTopicOverride({
        });
        throw new WrappedError('createStudentTopicOverride Empty object');
    } catch (e) {
    }

    /* *************** *************** */
    /* *************** *************** */
    /* *************** *************** */
    try {
        await courseRepository.createStudentTopicQuestionOverride({
            courseTopicQuestionId: 1,
            userId: 1,
            maxAttempts: 5
        });
    } catch (e) {
        throw new WrappedError('createStudentTopicQuestionOverride success', e);
    }
    try {
        await courseRepository.createStudentTopicQuestionOverride({
            courseTopicQuestionId: 1,
            userId: 1,
        });
    } catch (e) {
        throw new WrappedError('createStudentTopicQuestionOverride success min', e);
    }

    try {
        await courseRepository.createStudentTopicQuestionOverride({
            courseTopicQuestionId: -1,
            userId: 1,
        });
        throw new WrappedError('createStudentTopicQuestionOverride bad fk 1');
    } catch (e) {
    }

    try {
        await courseRepository.createStudentTopicQuestionOverride({
            courseTopicQuestionId: 1,
            userId: -1,
        });
        throw new WrappedError('createStudentTopicQuestionOverride bad fk 2');
    } catch (e) {
    }

    try {
        await courseRepository.createStudentTopicQuestionOverride({
        });
        throw new WrappedError('createStudentTopicQuestionOverride Empty object');
    } catch (e) {
    }
```
constraints and default values worked as expected

### CODE REVIEW

File: src/database/migrations/20200924034242-student-overrides-setup.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        endDate: {
+          field: 'student_topic_override_end_date',
+          type: DataTypes.DATE,
+          allowNull: true,
+        },
+        deadDate: {
+          field: 'student_topic_override_dead_date',
+          type: DataTypes.DATE,
+          allowNull: true,
+        },
```
Comment (doombot-exe) [inline src/database/migrations/20200924034242-student-overrides-setup.ts:50]: while we're at it, I suppose we might as well let profs override the startDate as well? I think that may be useful for a makeup exam, perhaps?

File: src/database/models/student-topic-override.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    endDate: {
+        field: 'student_topic_override_end_date',
+        type: DataTypes.DATE,
+        allowNull: true,
+    },
+    deadDate: {
+        field: 'student_topic_override_dead_date',
+        type: DataTypes.DATE,
+        allowNull: true,
+    },
```
Comment (doombot-exe) [inline src/database/models/student-topic-override.ts:83]: same

**Review by doombot-exe — CHANGES_REQUESTED** [review]

File: src/database/migrations/20200924034242-student-overrides-setup.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+          allowNull: false,
+        },
+        createdAt: {
+          field: 'created_at',
+          type: DataTypes.DATE,
+          allowNull: false,
+        },
+      });
+    });
+    /* eslint-disable @typescript-eslint/no-use-before-define */
```
Comment (PanopticaRising) [inline src/database/migrations/20200924034242-student-overrides-setup.ts:126]: Is this an artifact?

**Review by PanopticaRising — APPROVED** [review]

File: src/database/migrations/20200924034242-student-overrides-setup.ts (hunk 1 above)
Comment (PanopticaRising) [reply src/database/migrations/20200924034242-student-overrides-setup.ts:50]: I think that makes sense, if you specifically don't want a student to start something early.

File: src/database/migrations/20200924034242-student-overrides-setup.ts (hunk 3 above)
Comment (red-tt-bear) [reply src/database/migrations/20200924034242-student-overrides-setup.ts:126]: Good catch

File: src/database/migrations/20200924034242-student-overrides-setup.ts (hunk 1 above)
Comment (red-tt-bear) [reply src/database/migrations/20200924034242-student-overrides-setup.ts:50]: 🙄 now you tell me 😆

Comment (red-tt-bear) [reply src/database/migrations/20200924034242-student-overrides-setup.ts:55]: Resolved ^

**Review by doombot-exe — APPROVED** [review]

**Pull request merged**